chanmqttproxy/channelsmqttproxy.py

```diff
@@ -5,24 +5,76 @@
 import os
 import signal
 import socket
+import ssl
 
 from gmqtt import Client as MQTTClient
+from gmqtt.mqtt.handler import MQTTConnectError
 from gmqtt.mqtt.constants import MQTTv311, MQTTv50
 
 LOGGER = logging.getLogger(__name__)
 
 
 class ChannelsMQTTProxy:
+    @staticmethod
+    def strtobool(val):
+        """
+        FROM: https://stackoverflow.com/a/18472142
+        Convert a string representation of truth to true (1) or false (0).
+        True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
+        are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
+        'val' is anything else.
+        """
+        val = val.lower()
+        if val in ('y', 'yes', 't', 'true', 'on', '1'):
+            return True
+        elif val in ('n', 'no', 'f', 'false', 'off', '0'):
+            return False
+        else:
+            raise ValueError("invalid truth value %r" % (val,))
+
     def __init__(self, channel_layer, settings):
         self.channel_layer = channel_layer
 
         # MQTTClient takes an identifier which is seen at the broker
         # Creating the client does not connect.
         self.mqtt = MQTTClient(
             f"ChannelsMQTTProxy@{socket.gethostname()}.{os.getpid()}")
-        self.mqtt.set_auth_credentials(username=settings.MQTT_USER,
-                                       password=settings.MQTT_PASSWORD)
+        try:
+            self.mqtt.set_auth_credentials(username=settings.MQTT_USER,
+                                        password=settings.MQTT_PASSWORD)
+        except AttributeError:
+            # Settings are not defined. Try anonymous connection
+            pass
         self.mqtt_host = settings.MQTT_HOST
+        try:
+            self.mqtt_port = int(settings.MQTT_PORT)
+        except AttributeError:
+            # Setting not defined. Use default unsecured.
+            self.mqtt_port = 1883
+        # Set ssl
+        try:
+            self.mqtt_usessl = self.strtobool(settings.MQTT_USE_SSL)
```

Comment:
Did you test this? it seems broken.
settings.MQTT_USE_SSL is a bool already.

Please remove the strtobool() method and document it as being optional and taking True/False values.